pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -836,93 +837,45 @@ cdef int64_t parse_iso_format_string(str ts) except? -1:
     """
 
     cdef:
-        unicode c
         int64_t result = 0, r
-        int p = 0, sign = 1
-        object dec_unit = "ms", err_msg
-        bint have_dot = 0, have_value = 0, neg = 0
-        list number = [], unit = []
-
-    err_msg = f"Invalid ISO 8601 Duration format - {ts}"
-
-    if ts[0] == "-":
-        sign = -1
-        ts = ts[1:]
-
-    for c in ts:
-        # number (ascii codes)
-        if 48 <= ord(c) <= 57:
+        int sign = 1
+        object sign_part, integer_part, fractional_part
+        list groups = [], units = []
+
+    iso_regex = re.compile(
```

Comment:
Can you annotate this? I like the idea of using a regex here but this is hard to digest

Reply:
Sorry should be more clear on ask. I think you should use the `re.VERBOSE` flag and pair that with comments

Reply:
It is now broken up with `re.VERBOSE` and commented.